### PR TITLE
[HEI-264] Dev/architecture pre processing

### PR DESCRIPTION
**Pull request by TonyKatkov89**
Added pre-processing pipeline and tutorials for both filters and the pipeline. Also added fixes to filters, aggregator and rncoder.

### CODE REVIEW

**Review by Dmatryus — APPROVED** [review]
👌👌

File: hypex/encoders/abstract.py [hunk 1]
```diff
@@ -36,19 +41,19 @@ def _set_value(
...
+            executor_id=self._ids_to_names(value.columns),
+            name=str(self.__class__.__name__),
+            value=value,
             role=value.roles,
         )
 
     def execute(self, data: ExperimentData) -> ExperimentData:
         target_cols = data.ds.search_columns(
-            roles=self.target_roles or [FeatureRole()], search_types=[CategoricalTypes]
+            roles=self.target_roles, search_types=[CategoricalTypes]
```
Comment (VasilievaKA) [inline hypex/encoders/abstract.py:53]: u may add property for search types 
```python
@property 
def search_types(self): 
    return CategoricalTypes
```

File: hypex/transformers/category_agg.py [hunk 2]
```diff
@@ -41,7 +43,7 @@ def _inner_function(
         return data
 
     def execute(self, data: ExperimentData) -> ExperimentData:
-        target_cols = data.ds.search_columns(roles=FeatureRole())
+        target_cols = data.ds.search_columns(roles=self.target_roles, search_types=[CategoricalTypes])
```
Comment (VasilievaKA) [inline hypex/transformers/category_agg.py:46]: like previous comment

File: hypex/transformers/filters.py [hunk 3]
```diff
@@ -53,7 +55,7 @@ def _inner_function(
     def execute(self, data: ExperimentData) -> ExperimentData:
         if self.type_filter:
             target_cols = data.ds.search_columns(
-                roles=FeatureRole(), search_types=[float, int, bool]
+                roles=self.target_roles, search_types=[float, int, bool]
```
Comment (VasilievaKA) [inline hypex/transformers/filters.py:58]: like previous comment

**Review by VasilievaKA — APPROVED** [review]
Add some comments

File: hypex/encoders/abstract.py (hunk 1 above)
Comment (TonyKatkov89) [reply hypex/encoders/abstract.py]: done

File: hypex/transformers/filters.py (hunk 3 above)
Comment (TonyKatkov89) [reply hypex/transformers/filters.py]: done

File: hypex/transformers/category_agg.py (hunk 2 above)
Comment (TonyKatkov89) [reply hypex/transformers/category_agg.py]: done

**Pull request merged**